apps/server/src/modules/user-login-migration/service/restart-user-login-migration-validation.service.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { SchoolDO, UserLoginMigrationDO } from '@shared/domain';
+import { schoolDOFactory } from '@shared/testing';
+import { RestartUserLoginMigrationError } from '../error';
+import { RestartUserLoginMigrationValidationService } from './restart-user-login-migration-validation.service';
+import { CommonUserLoginMigrationService } from './common-user-login-migration.service';
+
+describe('RestartUserLoginMigrationValidationService', () => {
+	let module: TestingModule;
+	let service: RestartUserLoginMigrationValidationService;
+
+	let commonUserLoginMigrationService: DeepMocked<CommonUserLoginMigrationService>;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				RestartUserLoginMigrationValidationService,
+				{
+					provide: CommonUserLoginMigrationService,
+					useValue: createMock<CommonUserLoginMigrationService>(),
+				},
+			],
+		}).compile();
+
+		service = module.get(RestartUserLoginMigrationValidationService);
+		commonUserLoginMigrationService = module.get(CommonUserLoginMigrationService);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	describe('checkPreconditions', () => {
+		describe('when preconditions are met', () => {
+			const setup = () => {
+				const userId = 'userId';
+
+				const migration: UserLoginMigrationDO = new UserLoginMigrationDO({
```

Comment:
Please use the existing factory. You don't need to change you Testdata. Just use them as params

Reply:
I fixed that and pushed that, I don't know, why github did not mentioned it as outdated.

Reply:
I forgot to push the changes, now it's done